tls/test_tls_integrity.py

```diff
@@ -198,3 +198,66 @@ def test_various_req_resp_sizes(self):
         self.common_check(65536, 65536)
         self.common_check(1000000, 1000000)
 
+class CloseConnection(tester.TempestaTest):
+
+    clients = [
+        {
+            'id' : 'curl',
+            'type' : 'external',
+            'binary' : 'curl',
+            'cmd_args' : (
+                '-kfv -H "connection: close" ' # Set non-null return code on 4xx-5xx responses.
+                'https://${tempesta_ip}/ '
+                )
+        },
+    ]
+
+    backends = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'port' : '8000',
+            'response' : 'static',
+            'response_content' : 'HTTP/1.1 200 OK\r\n' \
+                'Content-Length: 65000\r\n' \
+                'Connection: close\r\n\r\n' +
+                'x' * 65000
+        }
+    ]
+
+    tempesta = {
+        'config' : """
+            cache 0;
+            listen 443 proto=https;
+            tls_certificate ${general_workdir}/tempesta.crt;
+            tls_certificate_key ${general_workdir}/tempesta.key;
+            server ${server_ip}:8000;
+        """
+    }
+
+    def start_all(self):
+        self.sniffer = analyzer.AnalyzerCloseRegular(self.get_tempesta().node,
+                                                     self.get_tempesta().host,
+                                                     node_close=True,
+                                                     timeout=10, ports=(443,))
+        self.sniffer.start()
+        #sleep(0.001) # Just schedule to another thread.
+
+        deproxy_srv = self.get_server('deproxy')
+        deproxy_srv.start()
+        self.start_tempesta()
+        self.deproxy_manager.start()
+        self.assertTrue(deproxy_srv.wait_for_connections(timeout=1),
+                        "No connection from Tempesta to backends")
+
+        curl = self.get_client('curl')
+        self.start_all_clients()
+        self.wait_while_busy(curl)
+
+    def tearDown(self):
+        tester.TempestaTest.tearDown(self)
+        self.sniffer.stop()
+        self.assertTrue(self.sniffer.check_results(), "Incorrect FIN-ACK sequence detected.")
```

Comment:
I see no reason to stop the sniffer after `teardown`. `self.wait_while_busy(curl)` ends when client connection is closed.

And interesting notice here, the test framework hungs if sniffer is stopped after `teardown()` and raises no issues and lasts for 11 seconds in total if this two lines are called after `start_all()`.